src/sentry/static/sentry/app/views/projectEvents.jsx

```diff
@@ -155,17 +150,14 @@ var ProjectEvents = React.createClass({
   },
 
   onPage(cursor) {
-    var router = this.context.router;
-    var params = router.getCurrentParams();
-    var queryParams = jQuery.extend({}, router.getCurrentQuery(), {
+    var queryParams = jQuery.extend({}, this.props.location.query, {
       cursor: cursor
     });
 
-    router.transitionTo('events', params, queryParams);
+    this.history.pushState(null, this.props.location.pathname, queryParams);
```

Comment:
Should be able to specify `null` in order to change query string w/o also specifying current URL:

https://developer.mozilla.org/en-US/docs/Web/API/History_API#The_pushState()_method